pandas/tests/indexing/test_interval.py

```diff
@@ -122,6 +148,46 @@ def test_with_slices(self):
         with pytest.raises(ValueError):
             s[0:4:2]
 
+    @pytest.mark.xfail(reason="new indexing tests for issue 16316")
+    def test_with_slices_updated_behavior(self):
+
+        s = self.s
+
+        # slice of interval
+        expected = s.iloc[4:]
+        result = s.loc[Interval(3, 4):]
+        tm.assert_series_equal(expected, result)
+
+        expected = s.iloc[4:]
+        result = s[Interval(3, 4):]
+        tm.assert_series_equal(expected, result)
+
+        with pytest.raises(KeyError):
+            s.loc[Interval(3, 6):]
+
+        with pytest.raises(KeyError):
+            s[Interval(3, 6):]
+
+        with pytest.raises(KeyError):
+            s.loc[Interval(3, 4, closed='left'):]
+
+        with pytest.raises(KeyError):
+            s[Interval(3, 4, closed='left'):]
+
+        with pytest.raises(KeyError):
+            s.loc[Interval(3, 4, closed='both'):]
+
+        with pytest.raises(KeyError):
+            s[Interval(3, 4, closed='both'):]
+
+        # slice of scalar
+        with pytest.raises(NotImplementedError):
+            s[0:4] ## not sure what the behvaior should be here.
```

Comment:
So you're suggesting that `df[:2]` and `df[:2.0]` have very different behaviors? This might be a bad idea.